src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlDataRecordTest.cs

```diff
@@ -318,6 +319,21 @@ public void GetChar_ThrowsNotSupported()
             Assert.Throws<NotSupportedException>(() => record.GetChar(0));
         }
 
+        [Theory]
+        [ClassData(typeof(GetUdtTypeTestData))]
+        public void GetUdt_ReturnsValue(Type udtType, object value, string serverTypeName)
+        {
+            SqlMetaData[] metadata = new SqlMetaData[] { new SqlMetaData(nameof(udtType.Name), SqlDbType.Udt, udtType, serverTypeName) };
+
+            SqlDataRecord record = new SqlDataRecord(metadata);
+
+            record.SetValue(0, value);
+
+            var checkValue = record.GetValue(0);
+
+            Assert.NotNull(checkValue);
```

Comment:
Can you also assert `checkValue` is the same as `value`?

Reply:
If I do that, the compiler starts complaining about a missing reference. Which I have no idea how to resolve.

Reply:
I kind of fixed it.